### PR TITLE
XML Config: Support for constant variable in level attributes (level, minlevel, etc)

### PR DESCRIPTION
**Pull request by 304NotModified**
allow ${loglevel} for minLevel, maxLevel, level and levels on `<logger>`

- [x] merge #2707 first
- [x] rebase then

fixes https://github.com/NLog/NLog/issues/2479

### CODE REVIEW

**Comment by 304NotModified** [comment]
maybe waiting first for https://github.com/NLog/NLog/pull/2624

File: src/NLog/Config/XmlLoggingConfiguration.cs [hunk 1]
```diff
@@ -576,7 +576,7 @@ private void ParseNLogElement(NLogXmlElement nlogElement, string filePath, bool
             }
 
             //check loglevel as first, as other properties could write (indirect) to the internal log.
-            InternalLogger.LogLevel = LogLevel.FromString(nlogElement.GetOptionalAttribute("internalLogLevel", InternalLogger.LogLevel.Name));
+            InternalLogger.LogLevel = LogLevelFromString(nlogElement.GetOptionalAttribute("internalLogLevel", InternalLogger.LogLevel.Name));
```
Comment (304NotModified) [inline src/NLog/Config/XmlLoggingConfiguration.cs]: unneeded?

File: src/NLog/Config/XmlLoggingConfiguration.cs [hunk 2]
```diff
@@ -600,7 +600,7 @@ private void ParseNLogElement(NLogXmlElement nlogElement, string filePath, bool
             InternalLogger.LogToTrace = nlogElement.GetOptionalBooleanAttribute("internalLogToTrace", InternalLogger.LogToTrace);
 #endif
             InternalLogger.IncludeTimestamp = nlogElement.GetOptionalBooleanAttribute("internalLogIncludeTimestamp", InternalLogger.IncludeTimestamp);
-            _logFactory.GlobalThreshold = LogLevel.FromString(nlogElement.GetOptionalAttribute("globalThreshold", _logFactory.GlobalThreshold.Name));
+            _logFactory.GlobalThreshold = LogLevelFromString(nlogElement.GetOptionalAttribute("globalThreshold", _logFactory.GlobalThreshold.Name));
```
Comment (304NotModified) [inline src/NLog/Config/XmlLoggingConfiguration.cs]: unneeded?

**Comment by 304NotModified** [comment]
docs added to https://github.com/NLog/NLog/wiki/configuration-file#variables-for-level-attributes

**Comment by snakefoot** [comment]
@304NotModified  I'm curious how to make use of this feature. Lets say you have this:

```xml
<nlog>
   <variable name='myLevel' value='debug'/>
    <rules>
      <logger minLevel='${myLevel}'/>
    </rules>
</nlog>
```

And you do this:

```
LogManager.KeepVariablesOnReload = true;
LogManager.Configuration.Variables["myLevel"] = "Info";
LogManager.Configuration.Reload();
```

Then nothing will happen because the old variables are applied after having parsed and reloaded the Nlog.config-file.

**Comment by snakefoot** [comment]
@304NotModified Created #3184 as possible solution for dynamic LogLevel-filtering

**Pull request merged**